sql/src/main/java/io/crate/action/sql/SessionContext.java

```diff
@@ -42,59 +43,72 @@
     private final StatementAuthorizedValidator statementAuthorizedValidator;
     private final ExceptionAuthorizedValidator exceptionAuthorizedValidator;
 
-    private String defaultSchema;
+    private SearchPath searchPath;
     private boolean semiJoinsRewriteEnabled = false;
     private boolean hashJoinEnabled = true;
 
     /**
      * Creates a new SessionContext suitable to use as system SessionContext
      */
     public static SessionContext systemSessionContext() {
-        return new SessionContext(null, User.CRATE_USER, s -> { }, e -> { });
+        return new SessionContext(0, Option.NONE, User.CRATE_USER, s -> { }, e -> { });
     }
 
-    public SessionContext(@Nullable String defaultSchema,
-                          User user,
+    public SessionContext(User user,
                           StatementAuthorizedValidator statementAuthorizedValidator,
-                          ExceptionAuthorizedValidator exceptionAuthorizedValidator) {
-        this(0, Option.NONE, defaultSchema, user,
-            statementAuthorizedValidator, exceptionAuthorizedValidator);
+                          ExceptionAuthorizedValidator exceptionAuthorizedValidator,
+                          @Nullable String... searchPath) {
+        this(0, Option.NONE, user,
+            statementAuthorizedValidator, exceptionAuthorizedValidator, searchPath);
     }
 
     public SessionContext(int defaultLimit,
                           Set<Option> options,
-                          @Nullable String defaultSchema,
                           User user,
                           StatementAuthorizedValidator statementAuthorizedValidator,
-                          ExceptionAuthorizedValidator exceptionAuthorizedValidator) {
+                          ExceptionAuthorizedValidator exceptionAuthorizedValidator,
+                          @Nullable String... searchPath) {
         this.defaultLimit = defaultLimit;
         this.options = options;
         this.user = requireNonNull(user, "User is required");
         this.statementAuthorizedValidator = statementAuthorizedValidator;
         this.exceptionAuthorizedValidator = exceptionAuthorizedValidator;
-        this.defaultSchema = defaultSchema;
-        if (defaultSchema == null) {
-            resetSchema();
+        this.searchPath = createSearchPathFrom(searchPath);
+    }
+
+    private static SearchPath createSearchPathFrom(String... schemas) {
```

Comment:
Would it make sense to have this function within the `SearchPath` ? It is not specific to the `SessionContext` and isn't introducing any other external dependencies